### PR TITLE
refactor: custom directions match a standard pattern

### PR DESCRIPTION
**Pull request by swalker125**
ParentsAndRespondentsDirections -> RespondentDirections

The benefits of this PR will be huge in FPL-894, when using custom direction collections to comply on behalf of others.

** Reverted changes to enum value, this should no longer be a breaking change **

**Does this PR introduce a breaking change?** (check one with "x")

```
[] Yes
[x] No
```

### CODE REVIEW

**Comment by swalker125** [comment]
@akomisarek if you would rather not have the breaking change I can change the ENUM back to PARENTS_AND_RESPONDENTS. Let me know.

File: service/src/main/java/uk/gov/hmcts/reform/fpl/enums/DirectionAssignee.java [hunk 1]
```diff
@@ -3,7 +3,7 @@
 public enum DirectionAssignee {
     ALL_PARTIES("allParties"),
     LOCAL_AUTHORITY("localAuthorityDirections"),
-    PARENTS_AND_RESPONDENTS("parentsAndRespondentsDirections"),
+    PARENTS_AND_RESPONDENTS("respondentDirections"),
```
Comment (Jamesnelson117) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/enums/DirectionAssignee.java]: Shall we rename this enum to `RESPONDENTS`?

**Comment by Jamesnelson117** [comment]
Why exactly is this needed @swalker125

File: service/src/main/java/uk/gov/hmcts/reform/fpl/enums/DirectionAssignee.java (hunk 1 above)
Comment (swalker125) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/enums/DirectionAssignee.java]: breaking change to cases in AAT and demo, I had it like that, but have reverted. Been open for a few days and no reviews so I thought the BREAKING CHANGE flag scared people off

**Comment by swalker125** [comment]
> Why exactly is this needed @swalker125

I want to be able to do put.(key.concat("Custom"), ...) without having an if to catch this case that doesn't match

File: service/src/main/java/uk/gov/hmcts/reform/fpl/enums/DirectionAssignee.java (hunk 1 above)
Comment (lewisbirks) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/enums/DirectionAssignee.java]: Can confirm was very scared 😨

**Pull request merged**